sesman/chansrv/clipboard_file.c

```diff
@@ -318,12 +349,20 @@ clipboard_send_file_size(int streamId, int lindex)
     if (g_files_list == 0)
     {
         LOG_DEVEL(LOG_LEVEL_ERROR, "clipboard_send_file_size: error g_files_list is nil");
+        clipboard_send_filecontents_response_fail(streamId);
         return 1;
     }
     cfi = (struct cb_file_info *)list_get_item(g_files_list, lindex);
     if (cfi == 0)
     {
         LOG_DEVEL(LOG_LEVEL_ERROR, "clipboard_send_file_size: error cfi is nil");
+        clipboard_send_filecontents_response_fail(streamId);
+        return 1;
+    }
+    if (cfi->size < 0)
+    {
+        LOG(LOG_LEVEL_ERROR, "clipboard_send_file_size: failed to get file size");
```

Comment:
Suggest reporting the value of the size here, so we can diagnose problems where the file is between 2 and 4GB in size. As we've been discussing [here](https://github.com/neutrinolabs/xrdp/discussions/1987) this is awkward for us at the moment.

Reply:
Thanks, I intended to catch the error of `stat` but indeed it will be recognized as a negative value if the size is between 2 and 4GB.
https://github.com/neutrinolabs/xrdp/blob/85518a81574e3153f266b4d66138f7a772450eee/common/os_calls.c#L2501-L2508